presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1498,7 +1491,39 @@ else if (expression instanceof LongLiteral) {
         return analyzedGroupingSets;
     }
 
-    private List<FieldOrExpression> analyzeGroupingColumns(List<Expression> groupingColumns, QuerySpecification node, RelationType tupleDescriptor, AnalysisContext context, List<FieldOrExpression> outputExpressions)
+    private List<Set<Expression>> computeGroupingSetsCrossProduct(List<List<Set<Expression>>> enumeratedGroupingSets, boolean isDistinct)
+    {
+        checkState(!enumeratedGroupingSets.isEmpty(), "enumeratedGroupingSets cannot be empty");
+
+        List<Set<Expression>> groupingSetsCrossProduct = new ArrayList<>();
+        groupingSetsCrossProduct.addAll(enumeratedGroupingSets.get(0)
+                        .stream()
+                        .map(ImmutableSet::copyOf)
+                        .collect(Collectors.toList()));
```

Comment:
instead of collecting to a list to insert into another list, you can use the forEach operation to add to the list